design/query-svc.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The Linux Foundation and each contributor to LFX.
+// SPDX-License-Identifier: MIT
+
+package design
+
+import (
+	"goa.design/goa/v3/dsl"
+)
+
+var JWTAuth = dsl.JWTSecurity("jwt", func() {
+	dsl.Description("Heimdall authorization")
+})
+
+var _ = dsl.Service("query-svc", func() {
+	dsl.Description("The query service provides resource and user queries.")
+
+	dsl.Error("BadRequest", BadRequestError, "Bad request")
+	dsl.Error("InternalServerError", InternalServerError, "Internal server error")
+	dsl.Error("ServiceUnavailable", ServiceUnavailableError, "Service unavailable")
+
+	dsl.Method("query-resources", func() {
+		dsl.Description("Locate resources by their type or parent, or use typeahead search to query resources by a display name or similar alias.")
+
+		dsl.Security(JWTAuth)
+
+		dsl.Payload(func() {
+			dsl.Extend(Sortable)
+			dsl.Token("bearer_token", dsl.String, func() {
+				dsl.Description("JWT token issued by Heimdall")
+				dsl.Example("eyJhbGci...")
+			})
+			dsl.Attribute("version", dsl.String, "Version of the API", func() {
+				dsl.Enum("1")
+				dsl.Example("1")
+			})
+			dsl.Attribute("name", dsl.String, "Resource name or alias; supports typeahead", func() {
+				dsl.Example("gov board")
+				dsl.MinLength(1)
+			})
+			dsl.Attribute("parent", dsl.String, "Parent (for navigation; varies by object type)", func() {
+				dsl.Example("project:123")
```

Comment:
We can add a regex here, I think. For example, we always expect this to be in the form `<resource>:<id>`, right?